src/api/instance/localInstanceImpl.js

```diff
@@ -0,0 +1,184 @@
+// @flow
+
+import {Instance} from "./instance";
+import {type CredrankInput} from "../credrank";
+import {
+  type WeightsT,
+  parser as weightsParser,
+  empty as emptyWeights,
+} from "../../core/weights";
+import fs from "fs-extra";
+import {join as pathJoin} from "path";
+import stringify from "json-stable-stringify";
+import {
+  type WeightedGraph,
+  fromJSON as weightedGraphFromJSON,
+} from "../../core/weightedGraph";
+import {
+  loadJson,
+  mkdirx,
+  loadFileWithDefault,
+  loadJsonWithDefault,
+} from "../../util/disk";
+import {parser as configParser, type InstanceConfig} from "../instanceConfig";
+import {Ledger} from "../../core/ledger/ledger";
+import {
+  parser as dependenciesParser,
+  type DependenciesConfig,
+} from "../dependenciesConfig";
+import {type Budget} from "../../core/mintBudget";
+import {parser as pluginBudgetParser} from "../pluginBudgetConfig";
+import {
+  CredGraph,
+  parser as credGraphParser,
+} from "../../core/credrank/credGraph";
+
+const DEPENDENCIES_PATH = ["config", "dependencies.json"];
+const WEIGHT_OVERRIDES_PATH = ["config", "weights.json"];
+const BUDGET_PATH = ["config", "pluginBudgets.json"];
+const INSTANCE_CONFIG_PATH = ["sourcecred.json"];
+const CREDGRAPH_PATH = ["output", "credGraph.json"];
+const GRAPHS_DIRECTORY = ["output", "graphs"];
+const GRAPHS_PATH = ["graph.json"];
+const LEDGER_PATH = ["data", "ledger.json"];
```

Comment:
These are global mutable state; since they’re not exported, can we at
least type them as `$ReadOnlyArray<string>` to prevent accidental
mutation?

Reply:
accepted